x-pack/legacy/plugins/ml/server/routes/data_frame_analytics.js

```diff
@@ -4,171 +4,298 @@
-    path: '/api/ml/data_frame/analytics/{analyticsId}/_stats',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
+  async function dfAnalyticsGetSingleJobStatsHandler(context, request, response) {
+    try {
       const { analyticsId } = request.params;
-      return callWithRequest('ml.getDataFrameAnalyticsStats', { analyticsId }).catch(resp =>
-        wrapError(resp)
-      );
-    },
-    config: {
-      ...commonRouteConfig,
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.getDataFrameAnalyticsStats', {
+        analyticsId,
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e });
+    }
+  }
+
+  router.get(
+    {
+      path: '/api/ml/data_frame/analytics/{analyticsId}/_stats',
+      validate: {
+        params: schema.object({ analyticsId: schema.string() }),
+      },
     },
-  });
+    licensePreRoutingFactory(xpackMainPlugin, dfAnalyticsGetSingleJobStatsHandler)
+  );
 
-  route({
-    method: 'PUT',
-    path: '/api/ml/data_frame/analytics/{analyticsId}',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
+  async function dfAnalyticsCreateJobHandler(context, request, response) {
+    try {
       const { analyticsId } = request.params;
-      return callWithRequest('ml.createDataFrameAnalytics', {
-        body: request.payload,
+      const results = await context.ml.mlClient.callAsCurrentUser('ml.createDataFrameAnalytics', {
+        body: request.body,
         analyticsId,
-      }).catch(resp => wrapError(resp));
-    },
-    config: {
-      ...commonRouteConfig,
-    },
-  });
-
-  route({
-    method: 'POST',
-    path: '/api/ml/data_frame/_evaluate',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      return callWithRequest('ml.evaluateDataFrameAnalytics', {
-        body: request.payload,
-      }).catch(resp => wrapError(resp));
-    },
-    config: {
-      ...commonRouteConfig,
-    },
-  });
-
-  route({
-    method: 'POST',
-    path: '/api/ml/data_frame/analytics/_explain',
-    handler(request) {
-      const callWithRequest = callWithRequestFactory(elasticsearchPlugin, request);
-      return callWithRequest('ml.estimateDataFrameAnalyticsMemoryUsage', {
-        body: request.payload,
-      }).catch(resp => wrapError(resp));
+      });
+      return response.ok({
+        body: { ...results },
+      });
+    } catch (e) {
+      // Case: default
+      return response.internalError({ body: e });
+    }
+  }
+
+  router.put(
+    {
+      path: '/api/ml/data_frame/analytics/{analyticsId}',
+      validate: {
+        params: schema.object({
+          analyticsId: schema.string(),
+        }),
+        body: schema.object({ ...dataAnalyticsJobConfigSchema }, { allowUnknowns: true }),
```

Comment:
Validation is our best protection against prototype pollution attacks, `allowUnknowns: true` disables a lot of that protection. Is it possible to enumerate all the expected properties in the schema?

Reply:
Good call - updated here: 66d9d97048c385885a158804899124f469390672